automata/base/automaton.py

```diff
@@ -108,5 +112,9 @@ def __repr__(self):
         """Return a string representation of the automaton."""
         values = ', '.join(
             f'{attr_name}={self._get_repr_friendly_value(attr_value)!r}'
-            for attr_name, attr_value in self.__dict__.items())
+            for attr_name, attr_value in self.attributes().items())
         return f'{self.__class__.__qualname__}({values})'
+
+    def __contains__(self, other):
```

Comment:
@Tagl Since `other` does not represent another automaton instance, but rather, the input string, could you rename `other` to `input_str` here? This would be more consistent with methods like `read_input_stepwise`.

Reply:
Updated